src/Sleep.js

```diff
@@ -1,26 +1,45 @@
 class Sleep {
-  constructor() {}
-  userSleepData(sleepData, id) {
-    this.userSleep = sleepData.filter(dailySleep => dailySleep.userID === id);
-    return this.userSleep;
+  constructor(sleepSet) {
+    this.sleepSet = sleepSet
   }
-  averageAllTimeSleep() {
-    let allTimeSleep = this.userSleep.reduce((sleep, day) =>{
-      sleep += day.sleepQuality;
-      return sleep 
+  userSleepData(id) {
+    return this.sleepSet.filter(dailySleep => dailySleep.userID === id);
+  }
+  averageAllTimeSleep(id) {
+    let getUserData = this.userSleepData(id)
+    let allTimeSleep = getUserData.reduce((sleep, day) =>{
+      return sleep += day.sleepQuality;
     }, 0)
-    return Number((allTimeSleep / this.userSleep.length).toFixed(1))
+    return Math.round((allTimeSleep / getUserData.length) * 10) / 10
+  }
+  daySleep(dateSelected, id) {
+    return this.sleepSet.find(day => day.date === dateSelected && day.userID === id).hoursSlept;
   }
-  daySleep(dateSelected) {
-    return this.userSleep.find(day => day.date === dateSelected).hoursSlept;
+  weeklySleepProperties(startDate, id, property) {
+    let startingDate = this.sleepSet.find(day => day.date === startDate && day.userID === id);
+    let firstDay = this.sleepSet.indexOf(startingDate);
+    return this.sleepSet.slice(firstDay, firstDay + 7).map(day => day[property])
+  }
```

Comment:
added a third parameter `property` here that allows us to return the both weekly `hoursSlept` and `sleepQuality` as seen by the bracket notation

Reply:
I like that we have made this more dynamic